app/custom-oauth/server/custom_oauth_server.js

```diff
@@ -225,6 +225,11 @@ export class CustomOAuth {
 					identity.id = identity.sub;
 				}
 
+				// Fix OpenShift identities where id is in 'metadata' object
+				if (identity.metadata.uid && !identity.id) {
```

Comment:
since going in a nested property you need to also ensure that identity.metadata is actually there.  Take a look at the other nested like for nextcloud below

Reply:
Ah, of course - fixed.